src/compiler/checker.ts

```diff
@@ -7418,7 +7413,7 @@ namespace ts {
                 if (links.jsxFlags & JsxFlags.ClassElement) {
                     let elemInstanceType = getJsxElementInstanceType(node);
 
-                    if (isTypeAny(elemInstanceType)) {
+                    if (!elemInstanceType || isTypeAny(elemInstanceType)) {

```

Comment:
Alternatively, you could return unknownType from getJsxElementInstanceType. The good thing about that is that then in the next line you could do
```ts
return links.resolvedJsxType = elemInstanceType;
```
You could then preserve what kind of `any` it was.